services/web3id-issuer/src/main.rs

```diff
@@ -268,13 +300,43 @@ async fn issue_credential(
         amount: Amount::zero(),
     };
 
+    let holder_id = request
+        .credential_subject
+        .id
+        .ty
+        .extract_public_key()
+        .ok_or(Error::InvalidId)?
+        .into();
+    if request.credential_subject.id.network != state.network {
+        return Err(Error::InvalidId);
```

Comment:
It should probably be  `return Err(Error::InvalidNetwork);` and `InvalidNetwork` should be added to the `Error` enum.

Reply:
Added.